### PR TITLE
Ensure OpenShift pod is terminated before returning from removeContainer

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Prevents issue where it is possible to start workspace while pod is terminating. When this occurs, Che creates a new pod but workspace start fails. This pod must be removed manually before that workspace can be launched again.

This is done by a) waiting for pod to terminate before returning from removeContainer, and b) removing created deployment and service when an exception occurs during createContainer.

Additionally, increases maximum wait time for retreiving OpenShift image stream metadata.

#### Changelog
- Fix issue where it was lose track of running workspace in OpenShift.

#### Docs PR
- No change to documentation, is a bugfix.

### CODE REVIEW

**Comment by amisevsk** [comment]
@ibuziuk If you have a moment, could you check if this PR helps with the `ImageStream not found` issue? I've stopped encountering it on this branch.

File: plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java [hunk 1]
```diff
@@ -967,6 +932,47 @@ private ContainerInfo createContainerInfo(Service svc,
...
+            LOG.info("Removing OpenShift Deployment {}", deployment.getMetadata().getName());
+            openShiftClient.resource(deployment).delete();
+        }
+
+        // Wait for all pods to terminate before returning.
+        List<Pod> pods = openShiftClient.pods()
+                                        .inNamespace(openShiftCheProjectName)
+                                        .withLabel(OPENSHIFT_DEPLOYMENT_LABEL, deploymentName)
+                                        .list()
+                                        .getItems();
```
Comment (l0rd) [inline plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java]: Why don't you just get the size of the list? And then iterate until this size get to zero? A for loop in a while loop complicated. You could avoid the for loop if you just get the size of the pods list at every iteration.

File: plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java [hunk 2]
```diff
@@ -967,6 +932,47 @@ private ContainerInfo createContainerInfo(Service svc,
...
+        try {
+            boolean podsTerminated = false;
+            while (!podsTerminated) {
```
Comment (l0rd) [inline plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java]: What if a pod never terminates? Shouldn't we wait OPENSHIFT_WAIT_POD_TIMEOUT and then exit?

**Review by l0rd — COMMENTED** [review]
Thank you @amisevsk for this PR. I've added 2 comments.

File: plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java (hunk 1 above)
Comment (amisevsk) [reply plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java]: That's a lot cleaner, thanks. Fixed.

File: plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java (hunk 2 above)
Comment (amisevsk) [reply plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java]: Fixed. Although, if a pod never terminates, the corresponding workspace will be blocked from launching, breaking a fair bit of Che.

**Review by l0rd — APPROVED** [review]

**Pull request merged**